pkg/network/tracer/tracer_linux_test.go

```diff
@@ -1695,3 +1699,101 @@ func TestPreexistingConnectionDirection(t *testing.T) {
 	assert.Equal(t, network.OUTGOING, conn.Direction)
 	assert.True(t, conn.IntraHost)
 }
+
+func TestEbpfConntrackerFallback(t *testing.T) {
+	var tests = []struct {
+		enableRuntimeCompiler    bool
+		allowPrecompiledFallback bool
+		rcError                  bool
+		prebuiltError            bool
+
+		err        error
+		isPrebuilt bool
+	}{
+		{false, false, false, false, nil, true},
+		{false, false, false, true, assert.AnError, false},
+		{false, false, true, false, nil, true},
+		{false, false, true, true, assert.AnError, false},
+		{false, true, false, false, nil, true},
+		{false, true, false, true, assert.AnError, false},
+		{false, true, true, false, nil, true},
+		{false, true, true, true, assert.AnError, false},
+		{true, false, false, false, nil, false},
+		{true, false, false, true, nil, false},
+		{true, false, true, false, assert.AnError, false},
+		{true, false, true, true, assert.AnError, false},
+		{true, true, false, false, nil, false},
+		{true, true, false, true, nil, false},
+		{true, true, true, false, nil, true},
+		{true, true, true, true, assert.AnError, false},
+	}
+
+	cfg := testConfig()
+	constants, err := getTracerOffsets(t, cfg)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		ebpfConntrackerPrebuiltCreator = getPrebuiltConntracker
+		ebpfConntrackerRCCreator = getRuntimeCompiledConntracker
+	})
+
+	for _, te := range tests {
+		t.Logf("%+v", te)
+		cfg.EnableRuntimeCompiler = te.enableRuntimeCompiler
+		cfg.AllowPrecompiledFallback = te.allowPrecompiledFallback
+
+		ebpfConntrackerPrebuiltCreator = getPrebuiltConntracker
+		ebpfConntrackerRCCreator = getRuntimeCompiledConntracker
+		if te.prebuiltError {
+			ebpfConntrackerPrebuiltCreator = func(c *config.Config, ce []manager.ConstantEditor) (bytecode.AssetReader, []manager.ConstantEditor, error) {
+				return nil, nil, assert.AnError
+			}
+		}
+		if te.rcError {
+			ebpfConntrackerRCCreator = func(cfg *config.Config) (rc.CompiledOutput, error) { return nil, assert.AnError }
+		}
+
+		conntracker, err := NewEBPFConntracker(cfg, nil, constants)
+		if te.err != nil {
+			assert.Error(t, err)
+			assert.Nil(t, conntracker)
+			continue
+		}
+
+		assert.NoError(t, err)
+		require.NotNil(t, conntracker)
```

Comment:
This failing will skip the rest of the tests, hence my suggestion to use `t.Run()` above.